include/plat/common/common_def.h

```diff
@@ -7,7 +7,6 @@
 #define __COMMON_DEF_H__
 
 #include <bl_common.h>
-#include <platform_def.h>
```

Comment:
I think this is not right. ``common_def.h`` uses ``BL2_BASE``, which is provided by ``platform_def.h`` so we have to maintain this inclusion.